### PR TITLE
✨ Generate `.shard_metadata` file in cron job shard

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: cron/controller/Dockerfile [hunk 1]
```diff
@@ -26,6 +26,8 @@ ARG TARGETARCH
 RUN CGO_ENABLED=0 make build-pubsub
 
 FROM gcr.io/distroless/base:nonroot@sha256:bc84925113289d139a9ef2f309f0dd7ac46ea7b786f172ba9084ffdb4cbd9490
+ARG COMMIT_SHA
+ENV SCORECARD_COMMIT_SHA=${COMMIT_SHA}
```
Comment (inferno-chromium) [inline cron/controller/Dockerfile:30]: Why need to redeclare, can just use COMMIT_SHA ?

File: cron/controller/main.go [hunk 2]
```diff
@@ -99,18 +102,37 @@ func main() {
...
+		NumShard:  new(int32),
+		ShardLoc:  new(string),
+		CommitSha: new(string),
+	}
+	*metadata.NumShard = (shardNum + 1)
+	*metadata.ShardLoc = bucket + "/" + data.GetBlobFilename("", t)
+	*metadata.CommitSha = os.Getenv(commitSHA)
+	metadataJSON, err := protojson.Marshal(&metadata)
+	if err != nil {
+		panic(err)
```
Comment (inferno-chromium) [inline cron/controller/main.go:132]: maybe add error message in these two panics ?

**Review by inferno-chromium — APPROVED** [review]

File: cron/controller/main.go (hunk 2 above)
Comment (azeemshaikh38) [reply cron/controller/main.go:132]: Done.

File: cron/controller/Dockerfile (hunk 1 above)
Comment (azeemshaikh38) [reply cron/controller/Dockerfile:30]: I tested it without ARG, and it doesn't seem to work. I guess we need to declare it for it to accept `--build-arg` option.

**Pull request merged**